exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -96,6 +98,32 @@ fun <T : Any?> ExpressionWithColumnType<T>.varPop(scale: Int = 2): VarPop<T> = V
  */
 fun <T : Any?> ExpressionWithColumnType<T>.varSamp(scale: Int = 2): VarSamp<T> = VarSamp(this, scale)
 
+// JSON Functions
+
+/**
+ * Returns the extracted data from a JSON object at the specified [path], either as a JSON representation or as a scalar value.
+ *
+ * @param path String(s) representing JSON path/keys that match fields to be extracted.
+ * **Note:** Multiple [path] arguments are not supported by all vendors; please check the documentation.
+ * @param toScalar If `true`, the extracted result is a scalar or text value; otherwise, it is a JSON object.
+ */
+inline fun <reified T : Any> ExpressionWithColumnType<*>.jsonExtract(vararg path: String, toScalar: Boolean = true): JsonExtract<T> {
+    val columnType = when (T::class) {
+        String::class -> TextColumnType()
+        Boolean::class -> BooleanColumnType()
+        Long::class -> LongColumnType()
+        Int::class -> IntegerColumnType()
+        Short::class -> ShortColumnType()
+        Byte::class -> ByteColumnType()
+        Double::class -> DoubleColumnType()
+        Float::class -> FloatColumnType()
```

Comment:
Should we add ByteArray as well?

Reply:
What should ByteArray be mapped to? `BlobColumnType`? Or maybe `BasicBinaryColumnType`, since the length is unknown?